index.js

```diff
@@ -135,51 +148,60 @@ function FSDuplexStream (path, opts) {
     this.setDefaultEncoding(opts.writeEncoding)
   }
 
-  this.path = path
-  this.bytesRead = 0
-  this.bytesWritten = 0
-
-  this._readEncoding = (opts.readEncoding || null)
+  Object.defineProperty(this, 'path', {
+    value: path,
+    enumerable: true
+  })
 
-  this._writeQueue = null
-  this._allIsRead = false
-  this._readTarget = 0
-  this._writeTarget = 0
+  this[AllIsRead] = false
+  this[BytesRead] = 0
+  this[BytesWritten] = 0
+  this[FinishCalled] = false
+  this[ReadEncoding] = (opts.readEncoding || null)
+  this[ReadTarget] = 0
+  this[WriteQueue] = null
+  this[WriteTarget] = 0
 
   fs.open(path, 'r+', onOpen.bind(this))
+}
 
-  var origEmit = this.emit
-  this.emit = function (ev) {
-    if (ev === 'finish') {
-      debug('intercepting finish event')
-
-      this.emit = origEmit
-      onFinish.call(this)
-
-      return
-    }
+FSDuplexStream.prototype = Object.create(Duplex.prototype)
```

Comment:
`Object.setPrototypeOf` is not available in Node 0.10 which we support, anyhow it won't actually affect this module since Duplex doesn't have any static properties that should be made available.